src/api/app/jobs/send_event_emails_job.rb

```diff
@@ -3,6 +3,7 @@ class SendEventEmailsJob < ApplicationJob
 
   def perform(event_id)
     event = Event::Base.find(event_id)
+    return if event.mails_sent
```

Comment:
in what scenario can a sendeventemailsjob exist but the event for it has mails sent? It's good old defensive programming, but I don't see how it helps in real life

Reply:
i am going to run a script in the console to update events.mails_sent = true if they have no subscribers i.e.
```ruby
count = 0
Event::Base.where(mails_sent: false).limit(1000).each do |event|
  if event.subscribers.empty?
    event.update_attributes(mails_sent: true)
    count += 1
  end
end

puts "#{count} events skipped." 
```

What do you think of that?

Reply:
oh - make that a delayed job! or revert #4340 :)